icechunk/src/change_set.rs

```diff
@@ -96,6 +96,10 @@ impl ChangeSet {
             .new_groups
             .iter()
             .filter(|(child_path, _)| child_path.starts_with(path))
+            // dcherian: sometimes we somehow end up deleting the same path twice
+            // I don't understand why, but hypothesis occasionally finds this behaviour
+            // It does not reproduce easily.
+            .filter(|(child_path, _)| child_path != &path)
```

Comment:
Here's a log that illustrates what can happen: (For some reason `"/5T/pE6Y.Q5O4dJ"` is deleted twice! the second time around that path is not in `new_groups` which means it gets added to `deleted_groups`)
```
Draw 6 (Group parent): '5T'
Adding group: path='5T/pE6Y.Q5O4dJ'
Checking 3 expected keys vs 3 actual keys
['zarr.json', '5T/zarr.json', '5T/pE6Y.Q5O4dJ/zarr.json']
['zarr.json', '5T/zarr.json', '5T/pE6Y.Q5O4dJ/zarr.json']
[icechunk/src/change_set.rs:63:9] &self = ChangeSet {
    new_groups: {
        Path(Utf8PathBuf {_encoding: "unix", inner: "/5T",},): f88bf7d7ac935e0d,
        Path(Utf8PathBuf {_encoding: "unix", inner: "/5T/pE6Y.Q5O4dJ",},): 66df7f9265bd04a8,
    },
    new_arrays: {},
    updated_arrays: {},
    updated_attributes: {
        f88bf7d7ac935e0d: Some(UserAttributes {parsed: Object {},},),
        66df7f9265bd04a8: Some(UserAttributes {parsed: Object {},},),
    },
    set_chunks: {},
    deleted_groups: {},
    deleted_arrays: {},
}
has uncommitted_changes: True

Draw 7 (Group deletion target): '5T'
Deleting group 'group_path='5T'', prefix='', group_name='5T' using delete
to delete: <Group <icechunk.store.IcechunkStore object at 0x106f4d9d0>>
[icechunk/src/change_set.rs:81:9] "deleting {} at {}" = "deleting {} at {}"
[icechunk/src/change_set.rs:81:9] &node_id = f88bf7d7ac935e0d
[icechunk/src/change_set.rs:81:9] &path = Path(
    Utf8PathBuf {_encoding: "unix", inner: "/5T",},
)
[icechunk/src/change_set.rs:81:9] "deleting {} at {}" = "deleting {} at {}"
[icechunk/src/change_set.rs:81:9] &node_id = 66df7f9265bd04a8
[icechunk/src/change_set.rs:81:9] &path = Path(Utf8PathBuf {_encoding: "unix", inner: "/5T/pE6Y.Q5O4dJ",},)
[icechunk/src/change_set.rs:81:9] "deleting {} at {}" = "deleting {} at {}"
[icechunk/src/change_set.rs:81:9] &node_id = 66df7f9265bd04a8
[icechunk/src/change_set.rs:81:9] &path = Path(Utf8PathBuf {_encoding: "unix", inner: "/5T/pE6Y.Q5O4dJ",},)
deleted   5T
to delete: <Group memory://4412194752>
deleted   5T
removed from self.all_groups which is now set()
Checking 1 expected keys vs 1 actual keys
['zarr.json']
['zarr.json']
[icechunk/src/change_set.rs:63:9] &self = ChangeSet {
    new_groups: {},
    new_arrays: {},
    updated_arrays: {},
    updated_attributes: {},
    set_chunks: {},
    deleted_groups: {
        (
            Path(Utf8PathBuf {_encoding: "unix", inner: "/5T/pE6Y.Q5O4dJ",},), 66df7f9265bd04a8,),
    },
    deleted_arrays: {},
}
has uncommitted_changes: True
[icechunk/src/change_set.rs:63:9] &self = ChangeSet {
    new_groups: {},
    new_arrays: {},
    updated_arrays: {},
    updated_attributes: {},
    set_chunks: {},
    deleted_groups: {
        (
            Path(Utf8PathBuf {_encoding: "unix", inner: "/5T/pE6Y.Q5O4dJ",},), 66df7f9265bd04a8,),
    },
    deleted_arrays: {},
}
```